src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleKey.java

```diff
@@ -79,35 +81,44 @@ public final String toString() {
    *
    * <p>This method must not be called if the module has a {@link NonRegistryOverride}.
    */
+  public RepositoryName getCanonicalRepoNameWithVersion(StarlarkSemantics semantics) {
+    return getCanonicalRepoName(/* includeVersion= */ true, semantics);
+  }
+
   public RepositoryName getCanonicalRepoNameWithVersion() {
```

Comment:
This overload seems a bit dangerous to keep around, what is it used for?

Reply:
it's used in tests. renamed to "XForTesting" for now.